src/value.rs

```diff
@@ -18,20 +18,28 @@ impl Value {
     pub fn str(val: &str) -> Value {
         Value::Str(val.to_owned())
     }
+}
 
-    pub fn is_truthy(&self) -> bool {
-        match *self {
-            Value::Bool(x) => x,
-            _ => true
-        }
-    }
+impl PartialEq<Value> for Value {
+    fn eq(&self, other: &Value) -> bool {
+        match (self, other) {
+            (&Value::Num(x), &Value::Num(y)) => x == y,
+            (&Value::Str(ref x), &Value::Str(ref y)) => x == y,
+            (&Value::Bool(x), &Value::Bool(y)) => x == y,
+            (&Value::Object(ref x), &Value::Object(ref y)) => x == y,
+            (&Value::Array(ref x), &Value::Array(ref y)) => x == y,
```

Comment:
I din't think so, the `x`s and `y`s are all different types, so I don't think the variable binding will work. I'll give it a go tonight, though, and see.